src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordConstructor.cs

```diff
@@ -5,59 +5,80 @@
 #nullable enable
 
 using System.Collections.Immutable;
+using System.Diagnostics;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CSharp.Symbols
 {
-    internal sealed class SynthesizedRecordConstructor : SynthesizedInstanceConstructor
+    internal sealed class SynthesizedRecordConstructor : SourceConstructorSymbolBase
     {
-        public override ImmutableArray<ParameterSymbol> Parameters { get; }
-
         public SynthesizedRecordConstructor(
-            SourceMemberContainerTypeSymbol containingType,
-            Binder parameterBinder,
-            ParameterListSyntax parameterList,
-            DiagnosticBag diagnostics)
-            : base(containingType)
-        {
-            Parameters = ParameterHelpers.MakeParameters(
-                parameterBinder,
-                this,
-                parameterList,
-                out _,
-                diagnostics,
-                allowRefOrOut: true,
-                allowThis: false,
-                addRefReadOnlyModifier: false);
-        }
-
-        internal override LexicalSortKey GetLexicalSortKey()
-        {
-            // We need a separate sort key because struct records will have two synthesized
-            // constructors: the record constructor, and the parameterless constructor
-            return LexicalSortKey.SynthesizedRecordCtor;
```

Comment:
>LexicalSortKey.SynthesizedRecordCtor; [](start = 19, length = 37)

Can we remove the sort key then (if no other reference)?

Reply:
> Can we remove the sort key then (if no other reference)?

Sure. I will do this in a following PR, unless a change is needed for this one.

---
In reply to: [434399175](https://github.com/dotnet/roslyn/pull/44803#discussion_r434399175) [](ancestors = 434399175)